CefSharp.Core/Internals/CookieManager.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright � 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#include "Stdafx.h"
+#include "CookieManager.h"
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        void CookieManager::ThrowIfDisposed()
+        {
+            if (_cookieManager.get() == nullptr)
+            {
+                throw gcnew ObjectDisposedException("CookieManager");
+            }
+        }
+
+        bool CookieManager::DeleteCookies(String^ url, String^ name)
```

Comment:
What do you think about renaming this to `DeleteCookiesAsync` and returning the actual task here?

Same for `SetCookie`